### PR TITLE
Superadmin view audit

### PR DESCRIPTION
**Pull request by Najaf**
Don't think this is ready to merge just yet. Here's some things I could do with a bit of advice on:

1. The tests are quite locked into the current HTML structure, any ideas on generalising these a little? It would be good if we could keep the order tested though.

2. Related to 1, I get the feeling that accessing the HTML directly isn't in keeping with the other tests. Any suggestions as to how to do these tests generally better would be appreciated.

3. `YAML::load` in the view!? My thinking is to pre-process the array of versions so that we effectively can do:

```ruby
# in a helper controller
def version_log_for(person)
  versions = VersionsCollectionPresenterOrSomething.new(person.versions)
  render 'audit', versions: versions
end
```

```haml
# in the partial
...
%ul
  - version.changes.each do |change|
    %li=change
```

If you don't like that, we can skip the helper and do the pre-processing in the controller instead.

Any of your thoughts re: the above or other problems appreciated.

Cheers,

@novotnyjakub, @cfq

### CODE REVIEW

**Comment by novotnyjakub** [comment]
I would maybe wrap the whole thing in a view object and test that properly. There you can cover various scenarios and then on feature level you just test that the versions are there? I prefer this approach to covering edge cases, etc. because it's much faster to run and maintain then feature tests. 

Also this view object would be possible to use later for displaying versions for a team, which is somewhere in the backlog..

**Comment by Najaf** [comment]
@novotnyjakub 

Here's the version presenter. Hopefully it's generic enough to use for a team as well as a person (in which case we can move the partial to `shared/`).

File: app/controllers/peoplefinder/people_controller.rb [hunk 1]
```diff
@@ -14,6 +14,7 @@ def index
 
     # GET /people/1
     def show
+      @versions = AuditVersionPresenter.wrap(@person.versions)
```
Comment (novotnyjakub) [inline app/controllers/peoplefinder/people_controller.rb]: Can we only load this when the super admin is logged in, rather than checking in the view?

File: app/views/peoplefinder/people/show.html.haml [hunk 2]
```diff
@@ -74,3 +74,7 @@
       = info_text('request_information')
       <br/>
       = link_to info_text('request_information_link'), new_person_information_request_path(@person)
+
+- if current_user.super_admin?
```
Comment (novotnyjakub) [inline app/views/peoplefinder/people/show.html.haml]: I'd remove this - check the comment above in the controller.

File: spec/features/person_audit_spec.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      person.update description: description
+      person.update primary_phone_number: phone_number
+    end
+  end
+
+  scenario 'View audit as an admin user' do
+    omni_auth_log_in_as(super_admin.email)
+    visit person_path(person)
+
+    expect(page).to have_selector 'table.audit tbody tr', count: 3
```
Comment (novotnyjakub) [inline spec/features/person_audit_spec.rb]: I'm not gonna push for it, but I introduced `SitePrism` to the project some time ago - to abstract html specifics from the feature specs. If you're interested and think it's a good idea, I'd move this to the page object. 
It may end up being a bit broader discussion about using page objects, of course.

**Comment by Najaf** [comment]
@novotnyjakub merge if you're happy.

**Comment by Najaf** [comment]
Don't merge this yet, it doesn't handle the case where the versions object_change contains an image upload.

**Comment by threedaymonk** [comment]
I think this is ready for merging now.

**Comment by Najaf** [comment]
Grudgingly happy to merge :neutral_face:

**Comment by threedaymonk** [comment]
Merged.
